### PR TITLE
Make marking materialized columns as MATERIALIZED stateless

### PR DESCRIPTION
**Pull request by macobo**
This simplifies the operational model a lot

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/clickhouse/materialized_columns/test/test_columns.py [hunk 1]
```diff
@@ -147,6 +147,18 @@ def test_backfilling_data(self, _):
             [("1", ""), ("2", "5"), ("3", ""), ("", ""), ("4", ""), ("", "6"), ("", "7")],
         )
 
+    def test_column_types(self):
+        materialize("events", "myprop")
+
+        expr = "replaceRegexpAll(JSONExtractRaw(properties, 'myprop'), concat('^[', regexpQuoteMeta('\"'), ']*|[', regexpQuoteMeta('\"'), ']*$'), '')"
```
Comment (neilkakkar) [inline ee/clickhouse/materialized_columns/test/test_columns.py:153]: Where is this expression coming from? O.o

Comment (macobo) [reply ee/clickhouse/materialized_columns/test/test_columns.py:153]: Clickhouse expands `trim(BOTH '\"' FROM JSONExtractRaw(properties, %(key)s))` to this in the table defs.

Comment (neilkakkar) [reply ee/clickhouse/materialized_columns/test/test_columns.py:153]: Worth a comment please?

**Review by neilkakkar — APPROVED** [review]
Makes sense, much nicer than before, given we don't have to backfill materialised columns ASAP once we run the command :)

**Pull request merged**